### PR TITLE
fix: logrotate for dragonfly logs

### LINKED ISSUE #1935: debian logrotate rule compresses the same files again and again
see `tools/packaging/debian/dragonfly.logrotate`
also, it compresses/rotates empty files though it should not.

as a result it produces the following files:
```
-rw-rw----  1 dfly adm            0 Jul 23 00:00 dragonfly.dragonflydb-production-02.dfly.log.ERROR.20230719-064229.1255
-rw-rw----  1 dfly adm            0 Jul 30 00:00 dragonfly.dragonflydb-production-02.dfly.log.ERROR.20230719-064229.1255.1.gz
-rw-rw----  1 dfly adm            0 Aug  6 00:00 dragonfly.dragonflydb-production-02.dfly.log.ERROR.20230719-064229.1255.1.gz.1.gz
-rw-rw----  1 dfly adm            0 Aug 13 00:00 dragonfly.dragonflydb-production-02.dfly.log.ERROR.20230719-064229.1255.1.gz.1.gz.1.gz
-rw-rw----  1 dfly adm            0 Aug 20 00:00 dragonfly.dragonflydb-production-02.dfly.log.ERROR.20230719-064229.1255.1.gz.1.gz.1.gz.1.gz
-rw-rw----  1 dfly adm            0 Aug 27 00:00 dragonfly.dragonflydb-production-02.dfly.log.ERROR.20230719-064229.1255.1.gz.1.gz.1.gz.1.gz.1.gz
-rw-rw----  1 dfly adm            0 Sep  3 00:00 dragonfly.dragonflydb-production-02.dfly.log.ERROR.20230719-064229.1255.1.gz.1.gz.1.gz.1.gz.1.gz.1.gz
-rw-rw----  1 dfly adm            0 Sep 10 00:00 dragonfly.dragonflydb-production-02.dfly.log.ERROR.20230719-064229.1255.1.gz.1.gz.1.gz.1.gz.1.gz.1.gz.1.gz
-rw-rw----  1 dfly adm            0 Sep 17 00:00 dragonfly.dragonflydb-production-02.dfly.log.ERROR.20230719-064229.1255.1.gz.1.gz.1.gz.1.gz.1.gz.1.gz.1.gz.1.gz
-rw-rw----  1 dfly adm            0 Sep 24 00:00 dragonfly.dragonflydb-production-02.dfly.log.ERROR.20230719-064229.1255.1.gz.1.gz.1.gz.1.gz.1.gz.1.gz.1.gz.1.gz.1.gz
-rw-rw----  1 dfly adm          847 Jul 19 07:59 dragonfly.dragonflydb-production-02.dfly.log.ERROR.20230719-064229.1255.1.gz.1.gz.1.gz.1.gz.1.gz.1.gz.1.gz.1.gz.1.gz.1.gz

```

### PR DESCRIPTION
**Pull request by romange**
The new logrotate settings assume that dragonfly closes a log file once it grows to large. It never rotates file that is currently open for writing.

Specifically logrotate:

1. rotate only log files
2. skip those that are currently open by a process.
3. compresses using zstd which is more cpu efficient than gzip
4. does not truncate/create old files as 0-sized blobs - just renames them

Fixes #1935

### CODE REVIEW

File: tools/packaging/debian/dragonfly.logrotate [hunk 1]
```diff
@@ -1,7 +1,26 @@
...
-        weekly
+# installed by debhelper by convention into /etc/logrotate.d/
+
+/var/log/dragonfly/dragonfly*.log {
+        daily
         missingok
-        rotate 12
+
         compress
+        compresscmd zstd
```
Comment (UkuLoskit) [inline tools/packaging/debian/dragonfly.logrotate:8]: In older versions of Debian `zstd` does not appear to be installed, so might need to add it as a dependency to the package.

Comment (romange) [reply tools/packaging/debian/dragonfly.logrotate:8]: Thanks! Do you know how to declare dependencies in debian package?

Comment (UkuLoskit) [reply tools/packaging/debian/dragonfly.logrotate:8]: I assume it's this: https://github.com/dragonflydb/dragonfly/blob/88852ccdd536d1b8b7141e0b9411837cb92f8f6e/tools/packaging/debian/control#L10

Comment (romange) [reply tools/packaging/debian/dragonfly.logrotate:8]: Thanks, will update! btw, I never asked you. Why did you decide to implement the slowlog feature?  Are you using Dragonfly or just for fun? Feel free to DM me on discord if you wish :)

Comment (UkuLoskit) [reply tools/packaging/debian/dragonfly.logrotate:8]: answered on Discord!

Comment (romange) [reply tools/packaging/debian/dragonfly.logrotate:8]: fixed dependency issue.

**Comment by adiholden** [comment]
@romange any chanes in helio are related to the logrotate? if not it might be better to seprate the PRs for updating helio

File: tools/packaging/debian/dragonfly.conf [hunk 2]
```diff
@@ -1,4 +1,5 @@
 --pidfile=/var/run/dragonfly/dragonfly.pid
 --log_dir=/var/log/dragonfly
 --dir=/var/lib/dragonfly
+--max_log_size=20
```
Comment (adiholden) [inline tools/packaging/debian/dragonfly.conf:4]: I dont find this flag.. where is it?

Comment (romange) [reply tools/packaging/debian/dragonfly.conf:4]: `./dragonfly --helpfull |  less` it's part of glog

**Review by adiholden — DISMISSED** [review]

**Comment by romange** [comment]
@ashotland  can you please review?

**Review by ashotland — APPROVED** [review]

**Pull request merged**